hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/fsck/ContainerHealthTask.java

```diff
@@ -185,6 +200,35 @@ private void processContainer(ContainerInfo container, long currentTime) {
     }
   }
 
+  private boolean containerDeletedInSCM(ContainerInfo containerInfo) {
+    try {
+      ContainerWithPipeline containerWithPipeline =
+          scmClient.getContainerWithPipeline(containerInfo.getContainerID());
```

Comment:
Is there a way we can maintain the state of a container as DELETED after getting the confirmation from SCM once? Since a container DELETED state is immutable, we do not need to keep asking the SCM if it is deleted repeatedly.

Reply:
I think `ContainerManager` is persisting the state already? When the event of DELETE and CLEANUP event was sent, the state changes should have been stored in Rocksdb?
```
          containerManager.updateContainerState(containerInfo.containerID(),
              HddsProtos.LifeCycleEvent.DELETE);
          containerManager.updateContainerState(containerInfo.containerID(),
              HddsProtos.LifeCycleEvent.CLEANUP);
```

Reply:
Got the point. Skipped `processContainer` for deleted containers.